desktop/cmp/input/CodeInput.js

```diff
@@ -179,28 +209,125 @@ export class CodeInput extends HoistInput {
                     ref: this.manageCodeEditor,
                     onChange: this.onChange
                 }),
-                this.renderActionButtons()
+                this.enableSearch ? this.renderToolbar() : this.renderActionButtons()
             ],
-
             className: this.getClassName(),
             onBlur: this.onBlur,
             onFocus: this.onFocus,
-
             ...props
         });
     }
 
+    renderToolbar() {
```

Comment:
This is a case where it would be a very good optimization to actually factor this out into its own sub-component, since it looks like this will re-render often.